File: backend/app/Models/Kampanye.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Kampanye extends Model
{
    protected $table = 'kampanye';
    public $timestamps = false;
    
    protected $fillable = [
        'judul',
        'deskripsi',
        'kategori',
        'target_dana',
        'dana_terkumpul',
        'url_gambar',
        'status',
        'dibuat_oleh'
    ];

    protected $casts = [
        'target_dana' => 'decimal:2',
        'dana_terkumpul' => 'decimal:2',
    ];

    protected $appends = ['persentase', 'sisa_dana'];

    public function getPersentaseAttribute()
    {
        if ($this->target_dana == 0) return 0;
        return round(($this->dana_terkumpul / $this->target_dana) * 100, 2);
    }

    public function getSisaDanaAttribute()
    {
        return max(0, $this->target_dana - $this->dana_terkumpul);
    }
}

File: lib/data/server/model/kampanye.dart
class Campaign {
  final int id;
  final String judul;
  final String deskripsi;
  final String kategori;
  final double targetDana;
  final double danaTerkumpul;
  final String? urlGambar;
  final String status;

  Campaign({
    required this.id,
    required this.judul,
    required this.deskripsi,
    required this.kategori,
    required this.targetDana,
    required this.danaTerkumpul,
    this.urlGambar,
    required this.status,
  });

  factory Campaign.fromJson(Map<String, dynamic> json) {
    return Campaign(
      id: json['id'] ?? 0,
      judul: json['judul'] ?? '',
      deskripsi: json['deskripsi'] ?? '',
      kategori: json['kategori'] ?? '',
      targetDana: double.parse(json['target_dana'].toString()),
      danaTerkumpul: double.parse(json['dana_terkumpul'].toString()),
      urlGambar: json['url_gambar'],
      status: json['status'] ?? 'aktif',
    );
  }

  double get persentase {
    if (targetDana == 0) return 0;
    return (danaTerkumpul / targetDana) * 100;
  }
}
